dffml/util/testing/consoletest/cli.py

```diff
@@ -43,17 +44,24 @@ async def main(argv: List[str]) -> None:
 
     nodes = []
 
-    for node in parse_nodes(args.infile.read()):
-        if not node.options.get("test", False):
-            continue
-        if node.directive == "code-block":
-            nodes.append(
-                code_block_to_dict(node.content, node.options, node=node.node)
-            )
-        elif node.directive == "literalinclude":
+    if args.code_block_content :
+
+        for node in parse_nodes(args.infile.read()):
+            if not node.options.get("test", False):
+                continue
+            if node.directive == "code-block":
+                nodes.append(
+                    code_block_to_dict(node.content, node.options, node=node.node)
+                )
+            elif node.directive == "literalinclude":
+                nodes.append(
+                    literalinclude_to_dict(node.content, node.options, node.node)
+                )
+    else :
+        for node in parse_nodes(args.infile.read()):
             nodes.append(
-                literalinclude_to_dict(node.content, node.options, node.node)
-            )
+                    code_block_to_dict(node.content, node.options, node=node.node)
+                )
```

Comment:
The issue says that we need to introduce functionality to parse input as if it was the contents of a code block.

```python
            nodes.append(
                code_block_to_dict(args.infile.read(), node.options, node=node.node)
            )
```